codersdk/users.go

```diff
@@ -104,15 +110,26 @@ type LoginWithPasswordResponse struct {
 	SessionToken string `json:"session_token" validate:"required"`
 }
 
+type OauthConversionResponse struct {
+	StateString string    `json:"state_string"`
+	ExpiresAt   time.Time `json:"expires_at" format:"date-time"`
+	ToLoginType LoginType `json:"to_login_type"`
+	UserID      uuid.UUID `json:"user_id" format:"uuid"`
+}
+
 type CreateOrganizationRequest struct {
 	Name string `json:"name" validate:"required,username"`
 }
 
 // AuthMethods contains authentication method information like whether they are enabled or not or custom text, etc.
 type AuthMethods struct {
-	Password AuthMethod     `json:"password"`
-	Github   AuthMethod     `json:"github"`
-	OIDC     OIDCAuthMethod `json:"oidc"`
+	// UserAuthenticationType returns the authentication method for the given
+	// caller if the request is an authenticated request. Otherwise it is empty.
+	UserAuthenticationType LoginType      `json:"me_login_type,omitempty"`
```

Comment:
The JSON field here doesn't match the struct field.

Reply:
@kylecarbs I did add it to the `/users/me` route, and can move it back. The downside was that I'd have to add it to `codersdk.Users`. If I did that, then all `/user(s)` calls would return the type with the login type.

But I do not want users to know other user's login types, and it becomes a security issue.

So I just punted this issue by putting it on this route. I think solving this security issue can be done better in it's own PR because I imagine there are other fields we would want to scope properly too in the future. 

----

1 solution is to make a `codersdk.DetailedUser` or something that is like:

```golang
type DetailedUser struct {
  User
  // Extra fields
  UserAuthenticationType LoginType
}
```

This becomes a new RBAC resource too.

Reply:
It being `me_login_type` was intentionally so it's clear to the caller that this is for `me` and only `me`. Not some other user.